VirtoCommerce.DynamicExpressionsModule.Data/Promotion/Conditions/CatalogConditions/ConditionEntryIs.cs

```diff
@@ -20,9 +24,18 @@ public class ConditionEntryIs : DynamicExpression, IConditionExpression
         {
             var paramX = linq.Expression.Parameter(typeof(IEvaluationContext), "x");
             var castOp = linq.Expression.MakeUnary(linq.ExpressionType.Convert, paramX, typeof(PromotionEvaluationContext));
-            var methodInfo = typeof(PromotionEvaluationContextExtension).GetMethod("IsItemInProduct");
+            linq.MethodCallExpression methodCall = null;
+            if (ProductIds != null)
+            {
+                var methodInfo = typeof(PromotionEvaluationContextExtension).GetMethod("IsItemInProducts");
+                methodCall = linq.Expression.Call(null, methodInfo, castOp, ProductIds.GetNewArrayExpression());
+            }
+            else if (!string.IsNullOrEmpty(ProductId))
+            {
+                var methodInfo = typeof(PromotionEvaluationContextExtension).GetMethod("IsItemInProduct");
+                methodCall = linq.Expression.Call(null, methodInfo, castOp, linq.Expression.Constant(ProductId));
+            }
 
-            var methodCall = linq.Expression.Call(null, methodInfo, castOp, linq.Expression.Constant(ProductId));
             var retVal = linq.Expression.Lambda<Func<IEvaluationContext, bool>>(methodCall, paramX);
```

Comment:
How this line would behave if methodCall == null? Will it cause exception in that case?

Reply:
Yes, it would be an exception. So need to remove condition from 'else' part,